### ISSUE TITLE
Select with object options puts "[object object]" in it's button's aria label

### ISSUE THREAD

**Issue by MikeKingdom** [grommet/grommet#7166]
Select with object options puts "[object object]" in it's button's aria label

### Expected Behavior

The aria-label on the `<button>` element should have the selected option's label

### Actual Behavior
The button ends up having `[object object]` in the aria-label. For example 
```html
<button id="select" aria-expanded="false" aria-haspopup="listbox" class="StyledButton-sc-323bzc-0 dkczdG StyledSelect__StyledSelectDropButton-sc-znp66n-5 eWjlvr"
  aria-label="Select; Selected: [object Object]" type="button">
```

### Steps to Reproduce
Modify the Select -> Object Options story to not have `reduce: true`
```jsx
import React, { useState } from 'react';

import { Box, Select } from 'grommet';

const objectOptions = [
  {
    label: 'Male',
    value: 1,
  },
  {
    label: 'Female',
    value: 2,
  },
  {
    label: 'Non Binary',
    value: 3,
  },
  {
    label: 'Other',
    value: 4,
  },
];

export const Aria = () => {
  const [value, setValue] = useState('');

  return (
    // Uncomment <Grommet> lines when using outside of storybook
    // <Grommet theme={...}>
    <Box fill align="center" justify="start" pad="large">
      <Select
        id="select"
        name="select"
        placeholder="Select"
        labelKey="label"
        valueKey="value"
        value={value}
        options={objectOptions}
        onChange={({ value: nextValue }) => setValue(nextValue)}
      />
    </Box>
    // </Grommet>
  );
};

Aria.storyName = 'Aria';

Aria.parameters = {
  chromatic: { disable: true },
};

export default {
  title: 'Input/Select/Aria',
};
```

### Your Environment

- Grommet version: 2.36.0
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): Windows 10 desktop

**Comment by jcfilben** [comment]
Duplicate of #6393, I'm going to close this issue just so that we have only one place to track this
